src/System.Private.CoreLib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -36,14 +39,15 @@ internal enum InsertionBehavior : byte
     [TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")]
     public class Dictionary<TKey, TValue> : IDictionary<TKey, TValue>, IDictionary, IReadOnlyDictionary<TKey, TValue>, ISerializable, IDeserializationCallback where TKey : notnull
     {
+        [StructLayout(LayoutKind.Auto)]
         private struct Entry
         {
+            public uint hashCode;
```

Comment:
Reordered the fields ( I assume by the order of access) and made layout Auto. Just curious if this was on purpose or a typo.

Reply:
On purpose; order of access, but also Auto in case the TKey and TValue are unusual size structs so the reordering doesn't work for the packing (e.g. `byte`, `byte`)

Reply:
If Auto considers the initial layout (I do not know), it would be a nice trick to learn.

Reply:
I wondered too, maybe worth a comment.